sheetScript.gs

```diff
@@ -66,91 +70,98 @@ function updatePlaylists(sheet) {
+	        var newChannelIds = getAllChannelIds();
+	        if (!newChannelIds || newChannelIds.length === 0) addError("Could not find any subscriptions");
+	        else [].push.apply(channelIds, newChannelIds);
+	      } else if (channel.substring(0,2) == "PL" && channel.length > 10)  // Add videos from playlist. MaybeTODO: better validation, since might interpret a channel with a name "PL..." as a playlist ID
+	         playlistIds.push(channel);
+	      else if (!(channel.substring(0,2) == "UC" && channel.length > 10)) // Check if it is not a channel ID (therefore a username). MaybeTODO: do a better validation, since might interpret a channel with a name "UC..." as a channel ID
+	      {
+	        try {
+	          var user = YouTube.Channels.list('id', {forUsername: channel, maxResults: 1});
+	          if (!user || !user.items) addError("Cannot query for user " + channel)
+	          else if (user.items.length === 0) addError("No user with name " + channel)
+	          else if (user.items.length !== 1) addError("Multiple users with name " + channel)
+	          else if (!user.items[0].id) addError("Cannot get id from user " + channel)
+	          else channelIds.push(user.items[0].id);
+	        } catch (e) {
+	          addError("Cannot search for channel with name "+channel+", ERROR: " + "Message: [" + e.message + "] Details: " + JSON.stringify(e.details));
+	          continue;
+	        }
+	      }
+	      else
+	        channelIds.push(channel);
+	    }
+	    
+	    /// ...get videos from the channels...
+	    var newVideoIds = [];
+	    for (var i = 0; i < channelIds.length; i++) {
+	      var videoIds = getVideoIdsWithLessQueries(channelIds[i], lastTimestamp)
+	      if (!videoIds || typeof(videoIds) !== "object") addError("Failed to get videos with channel id "+channelIds[i])
+	      else if (debugFlag_logWhenNoNewVideosFound && videoIds.length === 0) {
+	        Logger.log("Channel with id "+channelIds[i]+" has no new videos")
+	      } else {
+	        [].push.apply(newVideoIds, videoIds);
+	      }
+	    }
+	    for (var i = 0; i < playlistIds.length; i++) {
+	      var videoIds = getPlaylistVideoIds(playlistIds[i], lastTimestamp)
+	      if (!videoIds || typeof(videoIds) !== "object") addError("Failed to get videos with playlist id "+playlistIds[i])
+	      else if (debugFlag_logWhenNoNewVideosFound && videoIds.length === 0) {
+	        Logger.log("Playlist with id "+playlistIds[i]+" has no new videos")
+	      } else {
+	        [].push.apply(newVideoIds, videoIds);
+	      }
+	    }
+	      
+	    Logger.log("Acquired "+newVideoIds.length+" videos")
+	    
+	    if (!errorflag) {
+	      // ...add videos to playlist...
+	      if (!debugFlag_dontUpdatePlaylists) {
+	        addVideosToPlaylist(playlistId, newVideoIds);
+	      } else {
+	        addError("Don't Update Playlists debug flag is set");
+	      }
+	      
+	      /// ...delete old vidoes in playlist
+	      var daysBack = data[iRow][reservedColumnDeleteDays];
```

Comment:
One-off error since this is array index?

Reply:
In the Row or Column?

Reply:
column

Reply:
There was an off by one error before, that I corrected. It was still pointing to the old second column.

Reply:
Or am I mixing things up here?

Reply:
All of your constants are sheet column indexes, and any data array index should be `<constant> - 1` (like the `reservedColumnTimestamp - 1` = `reservedColumnPlaylist` indexing in findNextRow)

Reply:
I think you're right I apologize. Let me try and fix it.